### PR TITLE
validator registry view complete

### PR DESCRIPTION
**Pull request by askmish**
Signed-off-by: Ashish Kumar Mishra <ashish.k.mishra@intel.com>

### CODE REVIEW

File: src/core/src/validator_registry_view.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        sha.result_str()[..6].to_string()
+}
+
+fn _to_address(addressable_key: &String) -> String {
+    let mut sha = Sha512::new();
+    sha.input_str(addressable_key);
+    _vr_namespace_prefix() + &sha.result_str()[..64].to_string()
+}
+
+fn _as_validatorInfo(validatorInfoStr: String) -> ValidatorInfo {
```
Comment (rranjan3) [inline src/core/src/validator_registry_view.rs]: snake case for validatorInfoStr -> vlaidator_info_str

File: src/core/src/validator_registry_view.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            }
+        }
+
+        return result_map;
+    }
+
+    pub fn get_validator_info(&self, validator_id: &String) -> ValidatorInfo {
+        let validator_id_addr = _to_address(validator_id);
+        let state_data = self.state.get(validator_id_addr).unwrap();
+        return _as_validatorInfo(state_data);
```
Comment (rranjan3) [inline src/core/src/validator_registry_view.rs]: Why have the return keyword explicitly? Anything specific? Rust doesn't ask for it.

File: src/core/src/validator_registry_view.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    pub fn has_validator_info(&self, validator_id: &String) -> bool {
+        let validator_id_addr = _to_address(validator_id);
+        return self.state.contains_key(validator_id_addr);
```
Comment (rranjan3) [inline src/core/src/validator_registry_view.rs]: Here also

Comment (askmish) [reply src/core/src/validator_registry_view.rs]: This one is for bool

File: src/core/src/validator_registry_view.rs (hunk 2 above)
Comment (askmish) [reply src/core/src/validator_registry_view.rs]: This is just to make it consistent.

**Comment by askmish** [comment]
Done

**Pull request merged**